lib/evaluateFunctions.js

```diff
@@ -65,6 +68,229 @@ function evaluateFunctions(node) {
+
+      // 1. convert the number into the product of its prime factors
+      const factors = Factor.getPrimeFactors(radicandValue);
+      const factorNodes = factors.map(factor => {
+        return NodeCreator.constant(factor);
+      })
+
+      newNode.args[0] = NodeCreator.operator('*', factorNodes);
+      subSteps.push(NodeStatus.nodeChanged(
+          MathChangeTypes.FACTOR_INTO_PRIMES, oldNode, newNode));
+
+      // TODO: Consider moving steps 2, 3 and 4 into distributing the
+      // radical into children above, and just calling nthNode on the
+      // factored version
+
+      // 2. group any prime factors that show up root times
+      oldNode = NodeStatus.resetChangeGroups(newNode);
+      newNode = clone(oldNode, false);
+
+      let groupedFactors = [];
+      let hasGroups = false;
+      let i = 0;
+      while (i < factors.length) {
+        let j = i;
+        while (j < factors.length) {
+          if (factors[i] !== factors[j]) {
+            break;
+          }
+          j++;
+        }
+        if (j - i >= rootValue) {
+          let groupEnd = i + rootValue;
+          groupedFactors.push(factors.slice(i, groupEnd));
+          hasGroups = true;
+          i = groupEnd;
+        }
+        else {
+          groupedFactors = groupedFactors.concat(factors.slice(i, j));
+          i = j;
+        }
+      }
+
+      if (!hasGroups) {
+        return NodeStatus.noChange(node);
+      }
+
+      let children = [];
+      for (let i = 0; i < groupedFactors.length; i++) {
+        const item = groupedFactors[i];
+        if (groupedFactors[i].constructor === Array) {
+          const groupedChildren = [];
+          for (let j = 0; j < groupedFactors[i].length; j++) {
+            groupedChildren.push(
+              NodeCreator.constant(groupedFactors[i][j]));
+          }
+          children.push(NodeCreator.parenthesis(
+            NodeCreator.operator('*', groupedChildren)));
+        }
+        else {
+          children.push(NodeCreator.constant(groupedFactors[i]));
+        }
+      }
+
+      newNode.args[0] = NodeCreator.operator('*', children);
+      subSteps.push(NodeStatus.nodeChanged(
+          MathChangeTypes.GROUP_PRIME_FACTORS, oldNode, newNode));
+
+      // 3. convert grouped factors into exponent nodes
+      oldNode = NodeStatus.resetChangeGroups(newNode);
+      newNode = clone(oldNode, false);
+
+      for (let i = 0; i < children.length; i++) {
+        const child = children[i];
+        if (NodeType.isParenthesis(child)) {
+          const childNodeStatus = convertMultiplicationToExponent(child.content);
+          newNode.args[0].args[i] = childNodeStatus.newNode;
+          subSteps.push(NodeStatus.childChanged(newNode, childNodeStatus, i));
+        }
+      }
+
+      // 4. move grouped factors outside of the radical
+      oldNode = NodeStatus.resetChangeGroups(newNode);
+      newNode = clone(oldNode, false);
+
+      const newArgs = [];
+      const newRadicandArgs = [];
```

Comment:
confused about the difference between these two and why you always push to newRadicandArgs

comment examples would defs help